pkg/client/request.go

```diff
@@ -71,13 +138,16 @@ func (c *Client) Request(setters ...RequestOptionSetter) errors.Error {
 	rel := &url.URL{Path: opts.path}
 	u := c.config.APIDomain.ResolveReference(rel)
 
-	var buf io.ReadWriter
-	if opts.body != nil {
-		buf = new(bytes.Buffer)
-		err := json.NewEncoder(buf).Encode(opts.body)
-		if err != nil {
-			errors.NewClientInternalError().WithCause(err).Bug()
-		}
+	encoding, ok := mapEncodingTypeToEncoding[opts.BodyEncodingType]
+
+	if !ok {
+		return errors.NewClientInternalError().Bug()
```

Comment:
There should likely be a, like, `debug` mode we can put the CLI in to where we can surface some info. It'd likely be more idiomatic go to just panic here...

Reply:
As it happens, there is :) I was being lazy not using it. I added a cause with a non-zero sensitivity. It will only show up if the command is run with `--debug`